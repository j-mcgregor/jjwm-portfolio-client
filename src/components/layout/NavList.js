import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import NavLink from './NavLink';

const NavList = ({ closeModal, isAuthenticated, logoutUser }) => (
  <Fragment>
    <NavLink title="Home" onClick={closeModal} />
    <NavLink page="about" title="About" onClick={closeModal} />
    <NavLink page="portfolio" title="Portfolio" onClick={closeModal} />
    <NavLink page="skills" title="Skills" onClick={closeModal} />
    <NavLink page="experience" title="Experience" onClick={closeModal} />
    {!isAuthenticated ? (
      <NavLink page="auth" title="Auth" onClick={closeModal} />
    ) : (
      <NavLink page="auth" title="Logout" onClick={logoutUser} />
    )}
  </Fragment>
);

NavList.propTypes = {
  closeModal: PropTypes.func
};

NavList.defaultProps = {
  closeModal: () => {}
};

export default NavList;
