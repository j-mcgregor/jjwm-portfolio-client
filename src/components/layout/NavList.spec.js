import React from 'react';
import { shallow } from 'enzyme';
import NavList from './NavList';

describe('NavList', () => {
  it('should render', () => {
    const props = {
      closeModal: () => {}
    };
    const wrapper = shallow(<NavList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
