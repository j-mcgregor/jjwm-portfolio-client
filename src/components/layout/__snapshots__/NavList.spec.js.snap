// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavList should render 1`] = `ShallowWrapper {}`;
